src/itwinai/torch/loggers.py

```diff
@@ -0,0 +1,238 @@
+        """
+        if self._log_model == "all" or (
+            self._log_model is True and checkpoint_callback.save_top_k == -1
+        ):
+            self._scan_and_log_checkpoints(checkpoint_callback)
+        elif self._log_model is True:
+            self._checkpoint_callback = checkpoint_callback
+
+    def _scan_and_log_checkpoints(
+        self,
+        checkpoint_callback: ModelCheckpoint
+    ) -> None:
+        """Scans and logs checkpoints as artifacts in the experiment.
+
+        This function retrieves new checkpoints, logs them as artifacts, and saves
+        related metadata, including checkpoint score, filename, and any model checkpoint
+        parameters.
+
+        Args:
+            checkpoint_callback (ModelCheckpoint): Callback instance used for checkpointing.
+        """
+        # Get checkpoints to be saved with associated score
+        checkpoints = _scan_checkpoints(checkpoint_callback, self._logged_model_time)
+
+        # Log iteratively all new checkpoints
+        for t, p, s, tag in checkpoints:
+            metadata = {
+                "score": s.item() if isinstance(s, Tensor) else s,
+                "original_filename": Path(p).name,
+                checkpoint_callback.__class__.__name__: {
+                    k: getattr(checkpoint_callback, k)
+                    for k in [
+                        "monitor",
+                        "mode",
+                        "save_last",
+                        "save_top_k",
+                        "save_weights_only",
+                        "_every_n_train_steps",
+                    ]
+                    # ensure it does not break if `ModelCheckpoint` args change
+                    if hasattr(checkpoint_callback, k)
+                },
+            }
+            if p == checkpoint_callback.best_model_path:
+                aliases = ["latest", "best"]
+            else:
+                aliases = ["latest"]
+            artifact_path = Path(p).stem
+
+            # Log the checkpoint
+            self.experiment.log(
+                item=p,
+                identifier=artifact_path,
+                kind='artifact'
+            )
+
+            with tempfile.TemporaryDirectory(
+                prefix="test",
+                suffix="test",
+                dir=os.getcwd()
+            ) as tmp_dir:
+                # Log the metadata
+                with open(f"{tmp_dir}/metadata.yaml", "w") as tmp_file_metadata:
+                    yaml.dump(metadata, tmp_file_metadata, default_flow_style=False)
+
+                # Log the aliases
+                with open(f"{tmp_dir}/aliases.txt", "w") as tmp_file_aliases:
+                    tmp_file_aliases.write(str(aliases))
+
+                # Log the metadata and aliases
+                self.experiment.log(
+                    item=tmp_dir,
+                    identifier=artifact_path,
+                    kind='artifact'
+                )
```

Comment:
Are you sure that by passing the path to the directory all the files in it will be logged as artifacts? Is this working for all logging frameworks? If I well remember, by default they expect the path to a file. For instance MLFlow has a specific function ([`log_artifacts`](https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.log_artifacts)) to log all the artifacts in a folder.